### ISSUE TITLE
PI factor for theta in ggml_compute_forward_rope_f16

### ISSUE THREAD

**Issue by pierrel55** [ggerganov/llama.cpp#1234]
Hi,
This is more of a theoretical check than a bug report.

in ggml.c, function ggml_compute_forward_rope_f16()

                     const float cos_theta = cosf(theta);
                     const float sin_theta = sinf(theta);

It seems to me from the code that theta can only vary between 0 and 1.

Isn't there a PI factor to add somewhere ?

I apologize if my remark is useless, I do not know the theory of operation.

Regards.

**Comment by ggerganov** [comment]
This is implemented following the RoPE paper:

![image](https://user-images.githubusercontent.com/1991296/235315889-169d5913-bc09-4d77-9fd7-c68166fad294.png)

https://arxiv.org/pdf/2104.09864.pdf

I don't fully understand the logic of computing `theta` this way.
Would be interesting if somebody can ELI5

**Comment by pierrel55** [comment]
Thank you for the link.
So it may be normal that PI is not used. (It shouldn't be easy to figure out why.)
Best.
